### PR TITLE
Replace styfle/cancel-workflow-action with new GitHub `concurrency:` standard

### PR DESCRIPTION
**Pull request by vorburger**
See https://github.com/styfle/cancel-workflow-action/blob/main/README.md (and go/github-actions#cancel-workflows)

I've intentionally raised this separately from #2313 (which is included here) and #2314 (which builds on this) to make it easier to review, and revert in case it's needed because this unexpectedly breaks anything (which it should not).

This should be rebased after #2313 is merged.

### CODE REVIEW

**Comment by vorburger** [comment]
> Kokoro: Build and Device Tests — Kokoro build failed. Check logs for details.

due to #2297

**Comment by vorburger** [comment]
> Kokoro: Build and Device Tests — Kokoro build failed. Check logs for details.

OMG, it's #1482 again... how about we just do #2318 at this point?

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -34,6 +34,10 @@ env:
   GRADLE_BUILD_ACTION_CACHE_DEBUG_ENABLED: true
   GRADLE_BUILD_ACTION_CACHE_KEY_PREFIX: "fhir" # change this to invalidate cache
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref || github.run_id }}
```
Comment (jingtang10) [inline .github/workflows/build.yml:38]: if the run_id is included here in the group id, does this not mean that each run will have a separate group id? and does that mean previous runs will not be cancelled?

actually, if the head_ref is part of this, how would cancellation even help anything? shouldn't we cancel jobs that are in the same branch? i.e. if you push a new commit we should cancel the previous job?

Comment (jingtang10) [reply .github/workflows/build.yml:38]: thanks michael for the explanation.

this is clear to me now.

up to you about the run_id. but either way can we add a single line of comment to explain what's happening here? doesn't have to go through the length of your comment above, but just noting that we're using the branch/pr ref etc so possible to cancel previous runs.

**Review by jingtang10 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (jingtang10) [reply .github/workflows/build.yml:38]: once addressed please feel free to merge.

Comment (vorburger) [reply .github/workflows/build.yml]: Removed `github.run_id` and added inline documentation comments; I will self merge when build is green.

**Comment by vorburger** [comment]
FTR: I "tested" this by clicking the _Update Branch_ button, and seeing https://github.com/google/android-fhir/actions/runs/6696513296 got cancelled.

I'm going to merge this now, without waiting for Kokoro, because this is a GitHub-Action-only change, which cannot impact Firebase Test Lab runs (so Kokoro's result is irrelevant in this particular case).

**Pull request merged**